matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt

```diff
@@ -558,6 +528,21 @@ internal class TimelineChunk(
         }
     }
 
+    private fun validateInsertion(range: OrderedCollectionChangeSet.Range, results: RealmResults<TimelineEventEntity>): Boolean {
+        // Insertion can only happen from LastForward chunk after a sync.
+        if (isLastForward.get()) {
+            val firstBuiltEvent = builtEvents.firstOrNull()
+            if (firstBuiltEvent != null) {
+                val lastInsertion = results[range.startIndex + range.length - 1] ?: return false
+                if (firstBuiltEvent.displayIndex + 1 != lastInsertion.displayIndex) {
```

Comment:
for my understanding, is the only false case when the `firstBuiltEvent` exists and it doesn't have the same display index as the `results[range.startIndex + range.length - 1]`

it might be a little easier to read as a when rather than nested ifs 

```kotlin
private fun validateInsertion(range: OrderedCollectionChangeSet.Range, results: RealmResults<TimelineEventEntity>): Boolean {
    // Insertion can only happen from LastForward chunk after a sync.
    val firstForwardEvent = isLastForward.get().takeIf { it }?.let { builtEvents.firstOrNull()  }
    return when {
        firstForwardEvent == null -> true
        firstForwardEvent.nextDisplayIndexDoesNotMatch(results.last(range)) -> {
            Timber.v("There is no continuation in the chunk, chunk is not fully loaded yet, skip insert.")
            false
        }
        else -> true
    }
}

private fun RealmResults<TimelineEventEntity>.last(range: OrderedCollectionChangeSet.Range): TimelineEventEntity? {
    return this[range.startIndex + range.length - 1]
}

private fun TimelineEvent.nextDisplayIndexDoesNotMatch(event: TimelineEventEntity?): Boolean {
    return (displayIndex + 1) != event?.displayIndex
}
```